storage/src/main/java/tech/pegasys/teku/storage/server/kvstore/dataaccess/KvStoreCombinedDaoAdapter.java

```diff
@@ -553,6 +558,11 @@ public void addFinalizedState(final Bytes32 blockRoot, final BeaconState state)
       finalizedUpdater.addFinalizedState(blockRoot, state);
     }
 
+    @Override
+    public void deleteFinalizedState(final UInt64 slot) {
+      finalizedUpdater.deleteFinalizedState(slot);
+    }
+
```

Comment:
i do wonder if hte only interface we want is the deleteFinalizedState, and the rest is hidden and done for sanity

Reply:
I'm following similar approach used for addFinalizedState, imo it isn't too complicated to track down what changes are made when we store/delete a finalized state in the db atm.